### PR TITLE
KIN-411 improvements on product catalog page

### PR DESCRIPTION
**Pull request by CarlosMay7**
[KIN-411](https://loopcrack.atlassian.net/browse/KIN-411) Small [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=loopcrack-org&utm_campaign=badge_info)

### Problem

The product catalog page needs more dynamism when selecting tags and related items and also telling the user when there are no products available 

### Solution

Add dynamic subtitles on the sidebar and a message saying there are no products

### Requirements checklist

Include a list of requirements necessary to consider this pull request complete

- [x] It must only be shown the type of filter subtitle on the filters sidebar of the product catalog page when it is at least one selected.

- [x] There must be a message indicating that the catalog is empty when there are no product records or products that match the search bar search.

### Testing

Describe any tests (manual or automated) executed to verify your changes

- [x] Manually check when a category is deselected the subtitle disappears 
- [x] Manually check when a category tag is deselected the subtitle disappears 
- [x] Manually check when a product tag is deselected the subtitle disappears 

## Demo

image
image
image

[KIN-411]: https://loopcrack.atlassian.net/browse/KIN-411?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/Controllers/CtrlPublicPages.php [hunk 1]
```diff
@@ -105,6 +105,33 @@ public function viewCategory(): string
...
+        }
+
+        $emptyProductTags = true;
+
+        foreach ($productTags as $productTag) {
+            if ($productTag['selected']) {
+                $emptyProductTags = false;
+                break;
+            }
+        }
```
Comment (LUGAMAFE) [inline app/Controllers/CtrlPublicPages.php:133]:
```suggestion
        $selectedCategories = array_column($categories, 'selected');
        $emptyCategories = !in_array(true, $selectedCategories, true);

        $selectedCategoryTags = array_column($categoryTags, 'selected');
        $emptyCategoryTags = !in_array(true, $selectedCategoryTags, true);

        $selectedProductTags = array_column($productTags, 'selected');
        $emptyProductTags = !in_array(true, $selectedProductTags, true);
```

**Review by LUGAMAFE — APPROVED** [review]
Good job @CarlosMay7 just one nit please

File: app/Controllers/CtrlPublicPages.php (hunk 1 above)
Comment (Mauriciocr207) [reply app/Controllers/CtrlPublicPages.php:133]: Also I think this could be on the View, not in Controller. What do you think? @LUGAMAFE

**Review by FernandoJoachin — APPROVED** [review]

**Review by CarlosJCD — APPROVED** [review]

**Comment by CarlosJCD** [comment]
@CarlosMay7 please merge this PR.

File: app/Controllers/CtrlPublicPages.php (hunk 1 above)
Comment (LUGAMAFE) [reply app/Controllers/CtrlPublicPages.php]: I don't agree @Mauriciocr207

**Pull request merged**